### PR TITLE
Resolve typescript warnings for Rancher Components

### LINKED ISSUE #11977: Update typing for Rancher Components

### PR DESCRIPTION
**Pull request by rak-phillip**
### Summary
This resolves typescript warnings for Rancher Components.

Fixes #11977 

### Occurred changes and/or fixed issues

- Update `pkg/rancher-components/src/shim-vue.d.ts`
- Remove instances of `@ts-nocheck`
- Resolve each error flagged by tsc

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Review by cnotv — APPROVED** [review]
LGTM

**Comment by rak-phillip** [comment]
@cnotv this is ready for another review - we finally pass CI 🙃

**Comment by cnotv** [comment]
> @cnotv this is ready for another review - we finally pass CI 🙃

I was taking my time anyway 😅

File: pkg/rancher-components/src/components/Form/LabeledInput/LabeledInput.vue [hunk 1]
```diff
@@ -10,12 +9,16 @@ import { debounce } from 'lodash';
...
 
-declare module 'vue/types/vue' {
-  interface Vue {
-    onInput: (event: Event) => void | ((event: Event) => void);
-  }
+interface NonReactiveProps {
+  onInput: (event: Event) => void | ((event: Event) => void);
 }
 
+const provideProps: NonReactiveProps = {
```
Comment (cnotv) [inline pkg/rancher-components/src/components/Form/LabeledInput/LabeledInput.vue:16]: I apologize for asking after 4 E2E tests fail retry and the late question.
May I ask you why do we need these noops for the setup?

**Review by cnotv — APPROVED** [review]

File: pkg/rancher-components/src/components/Form/LabeledInput/LabeledInput.vue (hunk 1 above)
Comment (rak-phillip) [reply pkg/rancher-components/src/components/Form/LabeledInput/LabeledInput.vue:16]: The `noop` comment is a strategy to resolve the following tsc error and communicate that we expect this function to be empty at this point:

```
Unexpected empty method 'onInput'.eslint[@typescript-eslint/no-empty-function](https://github.com/typescript-eslint/typescript-eslint/blob/v5.4.0/packages/eslint-plugin/docs/rules/no-empty-function.md)
```

An alternative approach would be add `// eslint-disable-next-line @typescript-eslint/no-empty-function` above the offending function.

Comment (cnotv) [reply pkg/rancher-components/src/components/Form/LabeledInput/LabeledInput.vue:16]: We could disable the TS lint rule for `*.vue` files at this point and keep it undefined.

Comment (rak-phillip) [reply pkg/rancher-components/src/components/Form/LabeledInput/LabeledInput.vue:16]: It makes sense to consider, although such a change would be out of scope for this PR.

Comment (cnotv) [reply pkg/rancher-components/src/components/Form/LabeledInput/LabeledInput.vue:16]: Totally agree, which is why I did not ask it 😄

**Pull request merged**